app/grants/views.py

```diff
@@ -1339,6 +1339,8 @@ def grant_details(request, grant_id, grant_slug):
                 pk=grant_id
             )
 
+        if not grant.visible:
+            raise Grant.DoesNotExist
```

Comment:
^ I'm not very clear on why we have this here @owocki

Reply:
updated to Http404